hail/python/hailtop/utils/utils.py

```diff
@@ -532,3 +533,12 @@ async def __aexit__(self, exc_type, exc, tb):
         if self.threshold_ms is None or total > self.threshold_ms:
             self.timing['total'] = total
             log.info(f'{self.description} timing {self.timing}')
+
+
+def url_basename(url):
+    return os.path.basename(urllib.parse.urlparse(url).path)
+
+
+def url_join(url, path):
+    parsed = urllib.parse.urlparse(url)
```

Comment:
Why did you choose not to just use `urlib.parse.urljoin()`?

Reply:
urljoin takes a pair of URLs, url_join takes a url and a path.  I updated the docs to clarify.